### PR TITLE
Knowledge Distillation: add scaling to the distillation loss

### LINKED ISSUE #869: Knowledge Distillation: Scaling of the distillation loss
According to https://arxiv.org/pdf/1503.02531.pdf it is recommended to scale the distillation loss with T^2:

> Since the magnitudes of the gradients produced by the soft targets scale as 1/T 2
> it is important to multiply them by T^2 when using both hard and soft targets. This ensures that the
> relative contributions of the hard and soft targets remain roughly unchanged if the temperature used
> for distillation is changed while experimenting with meta-parameters.

There is currently **no** T^2 scaling in the [Keras Vanilla Knowledge DIstillation example](https://keras.io/examples/vision/knowledge_distillation/) 

To reflect on the content of the paper, I would suggest including the scaling described in the paper by changing the [line 110](https://github.com/keras-team/keras-io/blob/1caa8696350af88ef40ecef462fd82e429bf7b8a/examples/vision/knowledge_distillation.py#L110) from `)` to `) * self.temperature**2`.

### PR DESCRIPTION
**Pull request by katyagovorkova**
Add T**2 scaling to the distillation loss as suggested in the KD paper https://arxiv.org/abs/1503.02531

Closes #869

### CODE REVIEW

File: examples/vision/knowledge_distillation.py [hunk 1]
```diff
@@ -104,10 +104,20 @@ def train_step(self, data):
 
             # Compute losses
             student_loss = self.student_loss_fn(y, student_predictions)
+
+            """ Compute scaled loss from https://arxiv.org/abs/1503.02531
+                Since the magnitudes of the gradients produced by the soft
```
Comment (fchollet) [inline examples/vision/knowledge_distillation.py:109]: Please turn this into a comment (use `#`)

**Comment by fchollet** [comment]
@Kennethborup FYI

File: examples/vision/knowledge_distillation.py (hunk 1 above)
Comment (katyagovorkova) [reply examples/vision/knowledge_distillation.py]: Thanks for checking! Fixed now (also in `.ipynb` and `.md`).

**Comment by Kennethborup** [comment]
While I do not mind adding the scaling to closer reflect the content of the original paper, I want to point out that I intentionally left it out to simplify the example; the scaling has no effect on the distillation procedure itself except for how to choose the hyperparameters (weighting of losses and temperature), and using this particular scaling does not naïvely transfer to other distillation settings.

**Review by fchollet — APPROVED** [review]
